test/e2e/serial/tests/configuration.go

```diff
@@ -935,51 +1013,89 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+				klog.Infof("update Topology Manager Policy to 'single-numa-node' via the kubeletconfig owner %s/%s", ref.Kind, ref.Name)
+				err = fxt.Client.Get(ctx, client.ObjectKey{Name: ref.Name}, &performanceProfile)
 				Expect(err).ToNot(HaveOccurred())
 
-				// Here we're changing the topology manager policy to to single-numa-node
-				// after the scheduler has been deleted and restarted so now we can create a
-				// topology affinity error deployment to see if the deployment's pod will be pending or not
-				kcObj.TopologyManagerPolicy = v1beta1.SingleNumaNodeTopologyManagerPolicy
-				err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+				policy := v1beta1.SingleNumaNodeTopologyManagerPolicy
+				updatedProfile := performanceProfile.DeepCopy()
+				updatedProfile.Spec.NUMA = &perfprof.NUMA{
+					TopologyPolicy: &policy,
+				}
+				spec, err := json.Marshal(updatedProfile.Spec)
 				Expect(err).ToNot(HaveOccurred())
 
-				err = fxt.Client.Update(ctx, targetedKC)
-				g.Expect(err).ToNot(HaveOccurred())
-			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+				Expect(fxt.Client.Patch(ctx, updatedProfile,
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+					))).ToNot(HaveOccurred())
 
-			waitForMcpUpdate(fxt.Client, ctx, mcpsInfo, MachineConfig)
+				waitForMcpUpdate(fxt.Client, ctx, mcpsInfo, MachineConfig)
+			}
 
-			By("creating a topology affinity error deployment and check if the pod status is pending")
+			By("creating a Topology Affinity Error deployment and check if the pod status is pending")
 			deployment := createTAEDeployment(fxt, ctx, "testdp", serialconfig.Config.SchedulerName, cpuResources)
 
 			maxStep := 3
-			updatedDeployment := &appsv1.Deployment{}
+			updatedDeployment := appsv1.Deployment{}
 			for step := 0; step < maxStep; step++ {
 				time.Sleep(10 * time.Second)
-
 				By(fmt.Sprintf("ensuring the deployment %q keep being pending %d/%d", deployment.Name, step+1, maxStep))
-
-				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(deployment), updatedDeployment)
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(deployment), &updatedDeployment)
 				Expect(err).ToNot(HaveOccurred())
-
 				Expect(wait.IsDeploymentComplete(deployment, &updatedDeployment.Status)).To(BeFalse(), "deployment %q become ready", deployment.Name)
 			}
 
 			By("checking the deployment pod has failed scheduling and its at the pending status")
-			pods, err := podlist.With(fxt.Client).ByDeployment(ctx, *updatedDeployment)
+			pods, err := podlist.With(fxt.Client).ByDeployment(ctx, updatedDeployment)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(len(pods)).To(Equal(1))
+
+			podsPending := 0
+			for _, pod := range pods {
+				if pod.Status.Phase == corev1.PodPending {
+					podsPending++
+				}
+			}
+			Expect(podsPending).To(Equal(1))
```

Comment:
this should be in a different commit because it solves a different issue than the performance-profile adaptation

Reply:
in that commit scope please add an assertion on the pods' length it is not greater than 10 (so < 10)

Reply:
absolutely right will separate that into a different commit.